subprojects/core/src/integTest/groovy/org/gradle/api/invocation/GradleLifecycleAllprojectsIntegrationTest.groovy

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.invocation
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+
+class GradleLifecycleAllprojectsIntegrationTest extends AbstractIntegrationSpec {
+
+    def 'gradle.lifecycle.allprojects is applying before gradle.#api'() {
+        given:
+        settingsFile << """
+            rootProject.name = 'root'
+            gradle.lifecycle.allprojects { project ->
+                println "lifecycle.allprojects for \${project.name}"
+            }
+            gradle.$invocation { project ->
+                println "$api for \${project.name}"
+            }
```

Comment:
💭  As an additional demonstration of ordering I would swap the order in which the callbacks are registered.
I wouldn't make a difference for the test results, right?

Reply:
Yes, the results will remain the same

Reply:
Oops, sorry, we can't change order of the registration, since `gradle.allprojects`/`gradle.subprojects` is too eager